internal/provisioner/kops_provisioner_cluster.go

```diff
@@ -436,9 +436,17 @@ func (provisioner *KopsProvisioner) ProvisionCluster(cluster *model.Cluster, aws
 }
 
 // UpgradeCluster upgrades a cluster to the latest recommended production ready k8s version.
-func (provisioner *KopsProvisioner) UpgradeCluster(cluster *model.Cluster) error {
+func (provisioner *KopsProvisioner) UpgradeCluster(cluster *model.Cluster, awsClient aws.AWS) error {
 	logger := provisioner.logger.WithField("cluster", cluster.ID)
 
+	isAMIValid, err := awsClient.IsValidAMI(cluster.ProvisionerMetadataKops.AMI, logger)
+	if err != nil {
+		return errors.Wrapf(err, "error checking the AWS AMI Image %s", cluster.ProvisionerMetadataKops.AMI)
+	}
+	if !isAMIValid {
+		return errors.Wrapf(err, "invalid AWS AMI Image %s", cluster.ProvisionerMetadataKops.AMI)
```

Comment:
```suggestion
		return errors.Errorf("invalid AWS AMI Image %s", cluster.ProvisionerMetadataKops.AMI)
```
`err` should be nil here so no need to return it.

I see that I was the one that accidentally returned the `err` in the CreateCluster AMI check. Sorry about that. I will push up a PR to fix that.

Reply:
i can fix here as well

Reply:
done